test/ctia/task/update_mapping_test.clj

```diff
@@ -0,0 +1,133 @@
+(defn- update-mapping-stores!-test-helper
+  "If aliased? is true, test update-mapping-stores! with an aliased store.
+  If aliased? is false, test update-mapping-stores! with an unaliased store.
+  
+  This function builds up a `reduce` to functionally step through a sequence of
+  of stores, rollovers, and update-mapping-stores! calls, and test intermediate states.
+  "
+  [aliased?]
+  {:pre [(boolean? aliased?)]}
+  (let [; set up connection
+        store-properties (cond-> {:entity :incident
+                                  :indexname "ctia_incident"
+                                  :host "localhost"
+                                  :port 9200
+                                  :aliased aliased?}
+                           ; cheap trick to rollover store without adding docs
+                           aliased? (assoc :rollover {:max_docs 0}))
+
+        index-names (cond-> ["ctia_incident"]
+                      aliased? (conj "ctia_incident-write"))
+
+        ; minimal store (same shape as @ctia.store/stores)
+        [conn stores] (let [{:keys [conn] :as state} (init/init-es-conn! store-properties)]
+                        [conn {:incident [((:es-store incident/incident-entity)
+                                           state)]}])
+
+        testing-plan (gen-testing-plan 10)
+
+        ; update-mapping-stores! and rollover-stores should be able to run in
+        ; any order. the order is chosen below
+        fs (cond-> {:update-mapping-stores! task/update-mapping-stores!}
+             aliased?
+             (assoc
+               :rollover-stores
+               #(let [{:keys [nb-errors] :as responses} (rollover/rollover-stores %)]
+                  (testing "Rollover completed without errors"
+                    (is (= 0 nb-errors)
+                        (pr-str responses)))
+                  (testing ":incident store successfully rolled over"
+                    (is (get-in responses [:incident :rolled_over])
+                        (pr-str responses))))))
+
+        ; Store TestingStep -> Store
+        testing-fn (fn [stores {:keys [present absent add-field] :as _step_}]
+                     (let [chosen-order (shuffle (keys fs))
+                           stores (cond-> stores
+                                    add-field (assoc-in
+                                                [:incident 0 :state :config :mappings
+                                                 "incident" :properties (nth add-field 0)]
+                                                (nth add-field 1)))]
+                       (testing (str "Incides should correctly update with ordering " (vec chosen-order))
+                         (testing "Store should update without error"
+                           (run! (comp #(% stores) fs) chosen-order))
+                         (inspect-indices
+                           conn
+                           index-names
+                           (fn [index-name {:keys [mappings] :as _index_}]
```

Comment:
Unless you plan to reuse `inspect-indices`, I would put the content of that fn in `inspect-indices` to have all the performed checks in one place. I think that it would be easier to follow.